explainability-connectors/src/main/java/org/kie/trustyai/connectors/kserve/v2/KServeV2GRPCPredictionProvider.java

```diff
@@ -0,0 +1,186 @@
+package org.kie.trustyai.connectors.kserve.v2;
+
+import java.util.List;
+import java.util.concurrent.CompletableFuture;
+import java.util.stream.Collectors;
+import java.util.stream.IntStream;
+
+import org.kie.trustyai.connectors.kserve.v2.grpc.GRPCInferenceServiceGrpc;
+import org.kie.trustyai.connectors.kserve.v2.grpc.InferTensorContents;
+import org.kie.trustyai.connectors.kserve.v2.grpc.ModelInferRequest;
+import org.kie.trustyai.connectors.kserve.v2.grpc.ModelInferResponse;
+import org.kie.trustyai.connectors.utils.ListenableFutureUtils;
+import org.kie.trustyai.explainability.model.*;
+
+import com.google.common.util.concurrent.ListenableFuture;
+
+import io.grpc.ManagedChannel;
+import io.grpc.ManagedChannelBuilder;
+
+/**
+ * Wraps a KServe v2-compatible model server as a TrustyAI {@link PredictionProvider}
+ */
+public class KServeV2GRPCPredictionProvider implements PredictionProvider {
+
+    private static final String DEFAULT_TENSOR_NAME = "predict";
+    private static final KServeDatatype DEFAULT_DATATYPE = KServeDatatype.FP64;
+    private final String modelName;
+    private final ManagedChannel channel;
+    private final List<String> outputNames;
+
+    private KServeV2GRPCPredictionProvider(String target, String modelName, List<String> outputNames) {
+
+        this.modelName = modelName;
+        this.channel = ManagedChannelBuilder.forTarget(target)
+                .usePlaintext()
+                .build();
+        this.outputNames = outputNames;
+
+    }
+
+    /**
+     * Create a {@link KServeV2GRPCPredictionProvider} for a model with an endpoint at {@code target} and named {@code modelName}.
+     * In this case, the output names will be generated as {@code output-0}, {@code output-1}, ...
+     *
+     * @param target The remote KServe v2 model server
+     * @param modelName The model's name
+     * @return A {@link PredictionProvider}
+     */
+    public static KServeV2GRPCPredictionProvider forTarget(String target, String modelName) {
+        return KServeV2GRPCPredictionProvider.forTarget(target, modelName, null);
+    }
+
+    /**
+     * Create a {@link KServeV2GRPCPredictionProvider} with an endpoint at {@code target} and named {@code modelName}.
+     * In this case, the output names are specified with {@code outputNames}.
+     *
+     * @param target The remote KServe v2 model server
+     * @param modelName The model's name
+     * @param outputNames A {@link List} of output names to be used
+     * @return A {@link PredictionProvider}
+     */
+    public static KServeV2GRPCPredictionProvider forTarget(String target, String modelName, List<String> outputNames) {
+        return new KServeV2GRPCPredictionProvider(target, modelName, outputNames);
+    }
+
+    private void addFeature(InferTensorContents.Builder content, Feature feature) {
```

Comment:
I think it is fine to start with number/boolean only but it should be nice to expand it to cover more (most?) of the supported `Type` of TrustyAI

Reply:
I agree. We could merge this PR in order to test with Python first and then add missing feature types on a separate PR. wdyt?

Reply:
Fine for me 👍